data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpSinkResponseHandler.java

```diff
@@ -66,35 +63,28 @@ public HttpSinkResponseHandler(
    */
   @Override
   public Future<Void> handle(final HttpResponse<Buffer> response) {
-    // TODO if it's in structured, the SDK just calls getBytes on the response.body() which might lead to NPE.
-    MessageReader messageReader = VertxMessageFactory.createReader(response);
-    if (messageReader.getEncoding() == Encoding.UNKNOWN) {
+    try {
+      final var event = VertxMessageFactory.createReader(response).toEvent();
+      if (event == null) {
+        return Future.failedFuture(new IllegalArgumentException("event cannot be null"));
+      }
+
+      TracingSpan.decorateCurrent(vertx, event);
 
-      // When the sink returns a malformed event we return a failed future to avoid committing the message to Kafka.
+      return producer
+        .send(KafkaProducerRecord.create(topic, PartitionKey.extract(event), event))
+        .mapEmpty();
+
+    } catch (final Exception ex) {
       if (response.body() != null && response.body().length() > 0) {
+        // When the sink returns a malformed event we return a failed future to avoid committing the message to Kafka.
         return Future.failedFuture(
-          new IllegalResponseException("Unable to decode response: unknown encoding and non empty response")
+          new IllegalResponseException("Unable to decode response: unknown encoding and non empty response", ex)
         );
       }
 
       // Response is non-event, discard it
```

Comment:
Can we have a debug log here that states: "response is not recognized as event, discarding it"